### ISSUE TITLE
Crash in mate::Converter<const net::URLRequest*>::ToV8

### ISSUE THREAD

**Issue by miniak** [atom/electron#4591]
* Electron version: 0.36.8
* Operating system: Mac & Windows

The url_chain in **net::URLRequest** is can be empty. In this case, the **url()** method cannot be called as calling **vector<GURL>::back()** on an empty container is undefined.

**Comment by miniak** [comment]
This crash seems to be related to using URL scheme interceptors

**Comment by alregner** [comment]
Attaching related crash dump.
[Electron_2016-02-23-104047_Alberts-MacBook-Pro.crash.zip](https://github.com/atom/electron/files/144637/Electron_2016-02-23-104047_Alberts-MacBook-Pro.crash.zip)

**Comment by deepak1556** [comment]
Weird for `url_chain` to be empty when request object is alive, can you provide the example scenario that caused the crash ?

**Comment by miniak** [comment]
We have a quite complex app, where lots of requests are happening. The crash is not 100% reproducible. Also it happens 10x more on Mac than on Windows. We have built of local built of Electron to debug the issue. We could see in the debugger that url_chain size == 0.

**Comment by miniak** [comment]
But generally, we are doing this (error handling omitted):

```
protocol.interceptFileProtocol('file', function(request, callback) {
    var parts = url.parse(request.url);
    callback(path.join(app.getAppPath(), parts.pathname));
});
```

**Comment by miniak** [comment]
Could it be a race condition, maybe the **net::URLRequest** is modified on a different thread at the same time?

**Comment by alregner** [comment]
I found out that the request is in CANCELED state which is surprising - is the interceptor supposed to be called with canceled requests?

<img width="1280" alt="screen shot 2016-02-24 at 11 39 21 pm" src="https://cloud.githubusercontent.com/assets/14017241/13304051/ffc9a8c8-db52-11e5-8f0e-2a3f4faffa44.png">

**Comment by zcbenz** [comment]
Fixed by #4592.
